File: code_combat/7_Sarven_Desert/342-Pet_Adjutant/petAdjutant.js
// Your pet can help you survive until you can escape.

function onHear(event) {
    // event.message contains the text that was heard.
    // If somebody said "Fire"
    if (event.message == "Fire") {
        // Move to the bottom X mark with pet.moveXY()
        pet.moveXY(64, 16);
        // Say something with pet.say()
        pet.say("Meow fire!");
    }
    // If somebody said "Heal"
    else if (event.message == "Heal") {
        // Move to the top X mark with pet.moveXY()
        pet.moveXY(64, 52);
        // Say something with pet.say()
        pet.say("Meow help!");
    } 
}

pet.on("hear", onHear);

// You don't have to change the code below.
while(true) {
    var enemy = hero.findNearestEnemy();
    if (enemy) {
        // If an enemy is too strong.
        if (enemy.type == "brawler") {
            hero.say("Fire");
        } else {
            hero.attack(enemy);
        }
    } else { 
        // If your hero needs healing.
        if (hero.health < hero.maxHealth / 2) {
            hero.say("Heal");
        }
    }
}

File: projects/code_combat/7_Sarven_Desert/342-Pet_Adjutant/pet_adjutant.py
def onHear(event):
    if event.message == "Fire":
        pet.moveXY(64, 16)
        pet.say("Meow fire!")
    elif event.message == "Heal":
        pet.moveXY(64, 52)
        pet.say("Meow help!")


pet.on("hear", onHear)

while True:
    enemy = hero.findNearestEnemy()
    if enemy:
        if enemy.type == "brawler":
            hero.say("Fire")
        else:
            hero.attack(enemy)
    else:
        if hero.health < hero.maxHealth / 2:
            hero.say("Heal")
